src/engine/controls/ratecontrol.cpp

```diff
@@ -34,138 +34,167 @@ RateControl::RateControl(const QString& group,
+            ConfigKey(group, QStringLiteral("rate_perm_up_small")));
+    connect(m_pButtonRatePermUpSmall.get(),
+            &ControlObject::valueChanged,
+            this,
+            &RateControl::slotControlRatePermUpSmall,
             Qt::DirectConnection);
     m_pButtonRatePermUpSmall->setKbdRepeatable(true);
 
     // Temporary rate-change buttons
     m_pButtonRateTempDown =
-        new ControlPushButton(ConfigKey(group,"rate_temp_down"));
-    m_pButtonRateTempDownSmall =
-        new ControlPushButton(ConfigKey(group,"rate_temp_down_small"));
+            std::make_unique<ControlPushButton>(ConfigKey(group, QStringLiteral("rate_temp_down")));
+    m_pButtonRateTempDownSmall = std::make_unique<ControlPushButton>(
+            ConfigKey(group, QStringLiteral("rate_temp_down_small")));
     m_pButtonRateTempUp =
-        new ControlPushButton(ConfigKey(group,"rate_temp_up"));
-    m_pButtonRateTempUpSmall =
-        new ControlPushButton(ConfigKey(group,"rate_temp_up_small"));
-
-    // We need the sample rate so we can guesstimate something close
-    // what latency is.
-    m_pSampleRate = ControlObject::getControl(
-            ConfigKey(QStringLiteral("[App]"), QStringLiteral("samplerate")));
+            std::make_unique<ControlPushButton>(ConfigKey(group, QStringLiteral("rate_temp_up")));
+    m_pButtonRateTempUpSmall = std::make_unique<ControlPushButton>(
+            ConfigKey(group, QStringLiteral("rate_temp_up_small")));
 
     // Wheel to control playback position/speed
-    m_pWheel = new ControlTTRotary(ConfigKey(group, "wheel"));
+    m_pWheel = std::make_unique<ControlTTRotary>(ConfigKey(group, QStringLiteral("wheel")));
 
     // Scratch controller, this is an accumulator which is useful for
     // controllers that return individual +1 or -1s, these get added up and
     // cleared when we read
-    m_pScratch2 = new ControlObject(ConfigKey(group, "scratch2"));
+    m_pScratch2 = std::make_unique<ControlObject>(ConfigKey(group, QStringLiteral("scratch2")));
 
     // Scratch enable toggle
-    m_pScratch2Enable = new ControlPushButton(ConfigKey(group, "scratch2_enable"));
+    m_pScratch2Enable = std::make_unique<ControlPushButton>(
+            ConfigKey(group, QStringLiteral("scratch2_enable")));
     m_pScratch2Enable->set(0);
 
-    m_pScratch2Scratching = new ControlPushButton(ConfigKey(group,
-                                                            "scratch2_indicates_scratching"));
+    m_pScratch2Scratching = std::make_unique<ControlPushButton>(
+            ConfigKey(group, QStringLiteral("scratch2_indicates_scratching")));
     // Enable by default, because it was always scratching before introducing
     // this control.
     m_pScratch2Scratching->set(1.0);
 
-
-    m_pJog = new ControlObject(ConfigKey(group, "jog"));
-    m_pJogFilter = new Rotary();
+    m_pJog = std::make_unique<ControlObject>(ConfigKey(group, QStringLiteral("jog")));
+    m_pJogFilter = std::make_unique<Rotary>();
```

Comment:
this is a refactoring, right? Should we rebase on `main` instead? I know I changed this line in `main` in #13853 so there are bound to be conflicts...

Reply:
> this is a refactoring, right? Should we rebase on main instead?

Yes, and no: as metioned above, these are just preparational steps for the scratch fix #14059 which should go to 2.5 IMO